src/components/Inbox/Inbox.js

```diff
@@ -12,13 +12,36 @@ class Inbox extends Component {
     ]
   }
 
+  handleDelete = (index) => {
+    this.setState(prev => {
+      const newList = prev.messages.slice(0);
+      newList.splice(index, 1);
+      return { messages: newList };
+    });
+    console.log("delete handled")
+  }
+
+  handleMarkAsRead = (index, read) => {
+    this.setState(prev => {
+      const newList = prev.messages.slice(0);
+      const newItem = Object.assign({}, prev.messages[index]);
```

Comment:
The spread operator thing I mentioned elsewhere is very similar to Object.assign here. You could take advantage of assign here to merge in your new state!

```
Object.assign({}, prev.messages[index], {
    read: read
});
```